src/components/TextInput/TextInput.tsx

```diff
@@ -65,6 +86,44 @@ export type TextInputProps =
           obfuscated?: boolean;
       } & TextInputBaseProps);
 
+type ExtraActionButtonProps = {
+    extraAction: TextInputExtraAction;
+    disabled: boolean;
+    isFocusVisible: boolean;
+    focusProps: DOMAttributes<FocusableElement>;
+};
+
+const ExtraActionButton: ForwardRefRenderFunction<HTMLButtonElement | null, ExtraActionButtonProps> = (
+    { extraAction, disabled, isFocusVisible, focusProps },
+    ref,
+): ReactElement | null => {
+    if (!extraAction) {
+        return null;
+    }
+    return (
+        <button
+            className={merge([
+                'tw-flex tw-items-center tw-justify-center tw-transition-colors tw-rounded tw-p-1 -tw-mr-2',
+                disabled
+                    ? 'tw-cursor-default tw-text-text-disabled'
+                    : 'tw-text-text-weak hover:tw-bg-box-neutral-hover hover:tw-text-box-neutral-inverse-hover',
+                isFocusVisible && FOCUS_STYLE,
+            ])}
+            onClick={extraAction.onClick}
+            data-test-id="fondue-extra-action-icon"
+            aria-label={extraAction.title.toLowerCase()}
+            disabled={disabled}
```

Comment:
Wouldn't it be a good idea if the extra action itself could be disabled independently of the Input? So this would be something lie `disabled={disabled || extraAction.disabled}`?